src/test/java/com/squareup/javawriter/builders/TypeSpecTest.java

```diff
@@ -33,22 +36,96 @@
             .build())
         .build();
-
-    // TODO: fix modifiers
-    // TODO: fix annotations
-
     assertThat(toString(taco)).isEqualTo(""
         + "package com.squareup.tacos;\n"
         + "\n"
+        + "import java.lang.Override;\n"
         + "import java.lang.String;\n"
         + "\n"
         + "class Taco {\n"
-        + "  String toString() {\n"
+        + "  @Override public final String toString() {\n"
         + "    return \"taco\";\n"
         + "  }\n"
         + "}\n");
   }
 
+  @Test public void anonymousInnerClass() throws Exception {
+    ClassName foo = ClassName.create("com.squareup.tacos", "Foo");
+    ClassName bar = ClassName.create("com.squareup.tacos", "Bar");
+    ClassName thingThang = ClassName.create(
+        "com.squareup.tacos", ImmutableList.of("Thing"), "Thang");
+    ParameterizedTypeName thingThangOfFooBar
+        = ParameterizedTypeName.create(thingThang, foo, bar);
+    ClassName thung = ClassName.create("com.squareup.tacos", "Thung");
+    ClassName simpleThung = ClassName.create("com.squareup.tacos", "SimpleThung");
+    ParameterizedTypeName thungOfSuperBar
+        = ParameterizedTypeName.create(thung, WildcardName.createWithLowerBound(bar));
+    ParameterizedTypeName thungOfSuperFoo
+        = ParameterizedTypeName.create(thung, WildcardName.createWithLowerBound(foo));
+    ParameterizedTypeName simpleThungOfBar = ParameterizedTypeName.create(simpleThung, bar);
+
+    ParameterSpec thungParameter = new ParameterSpec.Builder()
+        .addModifiers(Modifier.FINAL)
+        .type(thungOfSuperFoo)
+        .name("thung")
+        .build();
+    TypeSpec aSimpleThung = new TypeSpec.Builder()
+        .supertype(simpleThungOfBar)
+        .anonymousTypeArguments("$N", thungParameter)
+        .addMethod(new MethodSpec.Builder()
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .name("doSomething")
+            .addParameter(bar, "bar")
+            .addCode("/* code snippets */\n")
+            .build())
+        .build();
+    TypeSpec aThingThang = new TypeSpec.Builder()
+        .supertype(thingThangOfFooBar)
+        .anonymousTypeArguments("")
+        .addMethod(new MethodSpec.Builder()
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(thungOfSuperBar)
+            .name("call")
+            .addParameter(thungParameter)
+            .addCode("return $L;\n", aSimpleThung)
+            .build())
+        .build();
+    TypeSpec taco = new TypeSpec.Builder()
+        .name(ClassName.create("com.squareup.tacos", "Taco"))
+        .addField(new FieldSpec.Builder()
+            .addModifiers(Modifier.STATIC, Modifier.FINAL)
+            .type(thingThangOfFooBar)
+            .name("NAME")
+            .initializer("$L", aThingThang)
+            .build())
+        .build();
+
+    // TODO: import Thing, and change references from "Thang" to "Thing.Thang"
```

Comment:
I don't think this is strictly necessary. It's one of those things a human writes because it knows `Thing.Thang` carries semantic value whereas a soulless (killing?) machine doesn't need it.